executor/docs/spec.md

```diff
@@ -0,0 +1,276 @@
+# Cook Executor
+
+The document outlines a specification for Cook's custom executor.
+
+It is a modular design allowing users to add functionality by writing reusable, single-responsibility processes. Declarative configuration would determine when then processes would be executed. An HTTP API exposed by the executor itself would enable communication between the executor and the tasks it starts.
+
+
+## Scheduler HTTP API
+
+The following enhancements to the existing Cook scheduler API will:
+
+1. Allow users to specify multiple commands when creating jobs
+2. Allow users to query the status code or codes of a completed job
+
+### Launch a Job
+
+It should be possible to specify multiple commands when launching a job. Any command failure will fail the whole job. In the future, it may be desirable to have finer-grained control over how failure will be handled (for example, ignoring failures from asynchronous monitoring commands).
+
+Request body schema additions:
+
+| JSON Key | Type | Description |
+|----------|------|-------------|
+| commands | array of command objects | the commands to run, as well as information about how they are to be run |
+
+`command` object schema:
+
+| JSON Key | Type | Description |
+|----------|------|-------------|
+| command  | string | the command to run |
+| async | boolean | a flag determining whether the command is run asynchronously.  defaults to `false` |
+| guard | boolean | a flag determining whether failure of the command will stop execution of the whole job. defaults to `false` |
+
+Example `commands`:
+
+
+    commands: [
```

Comment:
Can you include your current thinking on environment in the spec?

Have you thought about adding a way for an operator of cook to provide default (both mandatory and optional) commands to the list? This would allow things like:

1. Setting environment variables
2. default log4j processes